static/src/components/setting/index.vue

```diff
@@ -0,0 +1,387 @@
+<template>
+  <div class="stats-config row">
+    <div class="col-md-12">
+      <div
+        v-for="category in inputFields"
+        :key="category.display"
+        class="form-group row"
+      >
+        <div class="col-md-5">
+          <p> {{ category.display }} </p>
+        </div>
+        <div class="col-md-7 pull-right">
+          <div
+            v-for="field in category.fields"
+            :key="field.name"
+          >
+            <input
+              v-if="field.type==='TextField'"
+              v-model="externalConfigDict[field.name]"
+              type="text"
+              class="form-control input-sm"
+            >
+            <select
+              v-if="field.type==='SelectField'"
+              v-model="externalConfigDict[field.name]"
+              class="form-control input-sm"
+            >
+              <option
+                v-for="choice in field.choices"
+                :key="choice[0]"
+                :value="choice[0]"
+              >
+                {{ choice[1] }}
+              </option>
+            </select>
+          </div>
+        </div>
+      </div>
+      <div
+        v-if="privateInstance"
```

Comment:
in the past we tried avoiding coding based on particular instances, rather we tried thinking in terms of features that a user can enable. In this case, would it make sense to make the block visible only when the levels are configured? (e.g., something like `v-if=isNonEmpty(accessLevels)`)